daffodil-core/src/main/scala/org/apache/daffodil/dsom/CompiledExpression.scala

```diff
@@ -91,26 +130,45 @@ class ExpressionCompiler[T <: AnyRef] extends ExpressionCompilerBase[T] {
    * We don't want to allow turning on/off whether a format is delimited or
    * not based on runtime expressions, only what the delimiters are.
    */
-  def compileDelimiter(qn: NamedQName, staticNodeInfoKind: NodeInfo.Kind, runtimeNodeInfoKind: NodeInfo.Kind, property: Found,
+  def compileDelimiter(
+    qn: NamedQName,
+    staticNodeInfoKind: NodeInfo.Kind,
+    runtimeNodeInfoKind: NodeInfo.Kind,
+    property: Found,
     host: OOLAGHost): CompiledExpression[T] = {
+
     val isEvaluatedAbove = false
-    val expr = property.value
+    val exprOrLiteral = property.value
     val namespacesForNamespaceResolution = property.location.namespaces
     val compileInfoWherePropertyWasLocated = propertyCompileInfo(property)
-    val (exprForCompiling, isRealExpression) = exprOrLiteral(expr, staticNodeInfoKind, compileInfoWherePropertyWasLocated)
-    val compiled1 = compileExpression1(qn, staticNodeInfoKind, exprForCompiling, namespacesForNamespaceResolution, compileInfoWherePropertyWasLocated, isEvaluatedAbove, host,
-      isRealExpression)
-    if (compiled1.isConstant) return compiled1
-    if (staticNodeInfoKind == runtimeNodeInfoKind) return compiled1
-    //
-    // TODO: consider passing in a flag or some other way of avoiding this
-    // duplicate compile run.
-
-    // This is, this nodeInfo.Kind is used as the target type in the DPath expression compiler, and
-    //
-    val compiled2 = compileExpression1(qn, runtimeNodeInfoKind, exprForCompiling, namespacesForNamespaceResolution, compileInfoWherePropertyWasLocated,
-      isEvaluatedAbove, host, isRealExpression)
-    compiled2
+    val compiled1 = compileExpression(
+      qn,
+      staticNodeInfoKind,
+      exprOrLiteral,
+      namespacesForNamespaceResolution,
+      compileInfoWherePropertyWasLocated,
+      isEvaluatedAbove,
+      host)
+
+    if (compiled1.isConstant || (staticNodeInfoKind == runtimeNodeInfoKind)) {
+      compiled1
+    } else {
+      //
+      // TODO: consider passing in a flag or some other way of avoiding this
+      // duplicate compile run.
+
+      // This is, this nodeInfo.Kind is used as the target type in the DPath expression compiler, and
```

Comment:
Not sure what is going on with this comment line

Reply:
No idea, but it doesn't provide anything useful. WIll remove